### DISCUSSION TITLE
How to remove folders from the Quick Move and Recent lists?

### DISCUSSION THREAD

**Discussion by pto2k** [zanllp/infinite-image-browsing#348] [Q&A] [answered]
Hi,
Once added folders to the Quick Move list or the recent list, how to remove them? I can't find a way.

Thank you!

**Comment by zanllp** [comment, marked as answer]
<img width="450" alt="image" src="https://github.com/zanllp/sd-webui-infinite-image-browsing/assets/25872019/474cc0a4-0187-4b09-894e-096af4074c65">

**Comment by pto2k** [comment]
Thanks.. I did check that More menu but somehow failed to find it...  Upon further testing, I uncovered two factors which concealed it from me initially:

1. The 3 default folder (working folder/ home/ Desktop), I mistakenly thought I added them and try to remove. My bad. Apparently, they cannot be added or removed, which is acceptable.

2. I manually added a file path with capitalisation discrepancies, confusing the system. 
For instance, for the folder 'E:/Photo', I entered "e:/photo". It was accepted and appended to the list. However, when opened, only the 'Add to...' option showed in the 'More' menu. Using that, both 'E:/Photo' and "e:/photo" appeared in my list. Opening either only displays the 'Remove from..' option which removes only 'E:/Photo'.

(Additionally, I discovered that drive folders like 'E:' can be added to 'Quick Move' but produce an "E: not found" error upon use and redirect to the 'Root'.)

And about the **Recent** list, is there a way to clean it up?

Thank you.

**Reply by zanllp** [reply]
1. Some items in the QuickMove list are preset and cannot be deleted.
2. I haven't encountered these problems before. I will add some features later that will allow you to manage the QuickMove list directly on the homepage. Also, you cannot directly add a certain drive to the search path, as shown in the following picture.
![image](https://github.com/zanllp/sd-webui-infinite-image-browsing/assets/25872019/c4e94a01-f4a7-4182-aae8-c06d29758c6c)

3. The "recent" list is a historical record that currently does not provide any editing methods.But , i will add a "clear" button. It displays a maximum of 20 items. If you don't want a certain record to appear on top, you can open several unrelated pages to push it down, or directly edit the "recent" value in the "useGlobalStore" of the localStorage in the iib frontend.

**Reply by pto2k** [reply]
Thank you for considering these updates!

For the QuickMove list, please could you also consider a way to start 'Walk Mode' on one folder there directly?
I think the Walk Mode(aka Infinite Browsing) is one of the best feature and should be more accessible.

Thank you.

**Reply by zanllp** [reply]
One of the current plans is to add this feature. Originally, I planned to add it together last time, but the changes were too significant and I was afraid of problems. You can refer to the following picture to temporarily enable the "walk" mode.
<img width="642" alt="image" src="https://github.com/zanllp/sd-webui-infinite-image-browsing/assets/25872019/aabf18a7-dbfc-448d-8438-3743efb079e8">

**Reply by zanllp** [reply]
already added, https://github.com/zanllp/sd-webui-infinite-image-browsing/wiki/Change-log

**Reply by pto2k** [reply]
Thank you for the update!

Just for the record, I had 2 edge cases with the 'Remove' function on the Quick Move list.
I must have somehow added these yesterday during testing. Sorry I don't remember how I did it. 
Now I cleaned the iib.db file and can no longer reproduce it. Everything works fine.

![Snipaste_2023-07-30_12-55-07](https://github.com/zanllp/sd-webui-infinite-image-browsing/assets/4364117/51c87a58-f264-450b-879f-93e99118e50d)
- 'Remove' the '$img2img' will give a 'Bad Request' after confirming the 'Are you sure you want to delete?' dialog.
- 'Remove' the mysterious empty item will show the 'Remove Completed' pop up, but the item stays there.

**Reply by zanllp** [reply]
Deleting the iib.db file fixed the issue, because there were several corrupted data entries saved within it. In the new version of iib, img2img should not be able to be added or removed, as there is already a preset with the same path in the QuickMove list. Of course, this is all assuming that there are no bugs in the new version. 

Other edge cases will be dealt with when there is time.
